drivers/spi/spi_ll_stm32.meta.jinja2

```diff
@@ -0,0 +1,36 @@
+{# Copyright (c) 2018 Nordic Semiconductor ASA #}
+{# SPDX-License-Identifier: Apache-2.0         #}
+
+{% extends 'device_declare.jinja2' %}
+
+{% block init_params %}
+        {% do params.update({'irq_flag':'CONFIG_SPI_STM32_INTERRUPT'}) %}
+        {% do params.update({'config_struct':'spi_stm32_config'}) %}
+        {% do params.update({'data_struct':'spi_stm32_data'}) %}
+        {% do params.update({'api_struct':'api_funcs'}) %}
+        {% do params.update({'init_function':'spi_stm32_init'}) %}
+{% endblock %}
+
+{# Because these blocks are "scoped" all variables in the macro from the
+   base template can be accessed in these blocks #}
+{% block config_struct_body %}
+
+       .spi = (SPI_TypeDef *){{ "%x"|format(device['reg']['0']['address']['0']) }},
+       .pclken.bus = {{ device['clocks']['0']['bus'] }},
+       .pclken.enr = {{ device['clocks']['0']['bits'] }},
+#ifdef CONFIG_SPI_STM32_INTERRUPT
+       .irq_config = {{ irq_config_function_name }},
```

Comment:
It is not clear where "irq_config_function_name" is coming from.
For "data_struct_name", since we are populating a "data_struct", I can guess, even if maybe a data_struct.name would feel more "constructed" and so guessable.

But I don't see a "irq_config_function" function anywhere.

Reply:
Check the comment on line 14, the block is "scoped" and inherit all variables in parent macro (in device_declare.jinja2). The irq_config_function_name is created in the original device macro in the parent file.